lib/README.md

```diff
@@ -100,6 +102,14 @@ Wrapping code in `class_exists()` and `function_exists()` is usually inappropria
 
 When to use which prefix is a judgement call, but the general rule is that if you're unsure, use the `gutenberg` prefix because it will less likely give rise to naming conflicts.
 
+#### When not to use plugin-specific prefixes/suffixes
+
+The above recommendations in relation to plugin-specific prefixes/suffixes are relevant only to files in the `lib` directory and only in the Gutenberg plugin.
+
+`Gutenberg` prefixes/suffixes _should not_ be used in Core PHP code. When synching `/lib` files to Core, plugin-specific prefixes/suffixes are generally replaced with their `WP_` equivalents manually.
+
+Accordingly, you should avoid using plugin-specific prefixes/suffixes in any block PHP code. Core blocks in the plugin are [published as NPM packages](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/release.md#packages-releases-to-npm-and-wordpress-core-updates), which Core consumes as NPM dependencies. See [block naming conventions](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library#naming-convention-for-php-functions) for more information.
+
```

Comment:
Is it worth mentioning that if blocks code needs to call gutenberg versions of core functions, there's a transformation that can be used in our webpack config, to ensure that in Gutenberg the GB version is called, where in core, the core version is called?

https://github.com/WordPress/gutenberg/blob/00753a7698d1eae10999a2cf602a1b623fb17dd4/tools/webpack/blocks.js#L25-L36

It's also mentioned in this code comment:

https://github.com/WordPress/gutenberg/blob/00753a7698d1eae10999a2cf602a1b623fb17dd4/tools/webpack/blocks.js#L144-L148

Reply:
Good point. It would be good to surface that somewhere. 

I also wanted to update `block-library/README.md` to stress the points raised in this PR.

I thought `block-library/README.md` might be more appropriate since it'd live and travel with that package (?)

Reply:
> I thought block-library/README.md might be more appropriate since it'd live and travel with that package (?)

Yep, that'd be a better place for it!

Reply:
I should actually just bundle that with this PR

Reply:
Either way — we can always include it in both places 🙂